localstack-core/localstack/aws/protocol/service_router.py

```diff
@@ -179,13 +179,6 @@ def legacy_rules(request: Request) -> Optional[ServiceModelIdentifier]:
     method = request.method
     host = hostname_from_url(request.host)
 
-    # API Gateway invocation URLs
-    # TODO: deprecated with #6040, where API GW user routes are served through the gateway directly
-    if ("/%s/" % PATH_USER_REQUEST) in request.path or (
-        host.endswith(LOCALHOST_HOSTNAME) and "execute-api" in host
-    ):
-        return ServiceModelIdentifier("apigateway")
-
```

Comment:
Would it make sense to have a test which ensures that the handling of non-existing API GW routes works correctly?

Reply:
Yeah, that totally makes sense. And even this isn't properly being testable against AWS due to fundamental restrictions, I think a test would be super-valuable to (a) avoid any kind of regressions and most importantly (b) exactly define how the behavior should and does look like.

Reply:
Definitely agree! Will add them a bit later today